src/Curation/Curation.service.ts

```diff
@@ -0,0 +1,116 @@
+import { raw, SQL } from 'decentraland-server'
+import { hasAccess as hasCollectionAccess } from '../Collection/access'
+import { getMergedCollection } from '../Collection/utils'
+import { hasAccess as hasItemAccess } from '../Item/access'
+import { getMergedItem } from '../Item/utils'
+import { CollectionCuration } from './CollectionCuration'
+import { CurationType } from './Curation.types'
+import { ItemCuration } from './ItemCuration'
+
+export class CurationService<
+  T extends typeof CollectionCuration | typeof ItemCuration
+> {
+  public type: CurationType
+
+  // TODO: Type this better (or remove it completely and use+mock new CurationService)
```

Comment:
Isn't it possible to type it using generics? I'm not sure what you meant with this TODO.

Reply:
The problem here is the return type

```ts
static byType(type: CurationType): CurationService<any> {
```

to type it correctly, `CurationService` should be typed by `CurationType`, like `CurationService<CurationType.COLLECTION>`, but that's kinda hard given that the arg the constructor receives is a Model

```ts
export class CurationService<
  T extends typeof CollectionCuration | typeof ItemCuration
> {

  constructor(public CurationModel: T) {
```

I tried to instantiate `CurationService` using a `CurationType` at first but getting the return type of methods like `getLatestByIds` to be correct wasn't easy